applications/RANSApplication/custom_processes/rans_line_output_process.cpp

```diff
@@ -0,0 +1,516 @@
+{
+    KRATOS_TRY
+
+    rParameters.ValidateAndAssignDefaults(GetDefaultParameters());
+
+    mModelPartName = rParameters["model_part_name"].GetString();
+    mWriteHeader = rParameters["write_header_information"].GetBool();
+
+    const Vector& r_start_point = rParameters["start_point"].GetVector();
+    KRATOS_ERROR_IF(r_start_point.size() != 3)
+        << "\"start_point\" should be given as 3-valued vector. [ "
+           "start_point = "
+        << r_start_point << " ]\n";
+    const Vector& r_end_point = rParameters["end_point"].GetVector();
+    KRATOS_ERROR_IF(r_end_point.size() != 3)
+        << "\"end_point\" should be given as 3-valued vector. [ "
+           "end_point = "
+        << r_end_point << " ]\n";
+    noalias(mStartPoint) = r_start_point;
+    noalias(mEndPoint) = r_end_point;
+
+    mNumberOfSamplingPoints = rParameters["number_of_sampling_points"].GetInt();
+    KRATOS_ERROR_IF(mNumberOfSamplingPoints <= 2)
+        << "Sampling points should be greater than 2. [ "
+           "number_of_sampling_points = "
+        << mNumberOfSamplingPoints << " ]\n";
+
+    mVariableNames = rParameters["variable_names_list"].GetStringArray();
+    mIsHistoricalValue = rParameters["historical_value"].GetBool();
+
+    // output controls
+    mOutputFileName = rParameters["output_file_name"].GetString();
+    mOutputStepControlVariableName =
+        rParameters["output_step_control_variable_name"].GetString();
+    mOutputStepInterval = rParameters["output_step_interval"].GetInt();
+
+    mPreviousStepValue = 0.0;
+
+    KRATOS_CATCH("");
+}
+
+int RansLineOutputProcess::Check()
+{
+    KRATOS_TRY
+
+    const auto& r_model_part = mrModel.GetModelPart(mModelPartName);
+
+    if (mIsHistoricalValue) {
+        for (const auto& variable_name : mVariableNames) {
+            if (KratosComponents<Variable<double>>::Has(variable_name)) {
+                const auto& r_variable = KratosComponents<Variable<double>>::Get(variable_name);
+                KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(r_variable))
+                    << variable_name << " is not found in nodal solution step variables list of "
+                    << mModelPartName << ".";
+            } else if (KratosComponents<Variable<array_1d<double, 3>>>::Has(variable_name)) {
+                const auto& r_variable = KratosComponents<Variable<array_1d<double, 3>>>::Get(variable_name);
+                KRATOS_ERROR_IF(!r_model_part.HasNodalSolutionStepVariable(r_variable))
+                    << variable_name << " is not found in nodal solution step variables list of "
+                    << mModelPartName << ".";
+            } else {
+                KRATOS_ERROR
+                    << "Variable name = " << variable_name
+                    << " not found in double, 3d-double variable lists.\n";
+            }
+        }
+    }
+
+    return 0;
+
+    KRATOS_CATCH("");
+}
+
+void RansLineOutputProcess::ExecuteInitialize()
```

Comment:
As it is right now, this only works in an Eulerian framework. To make it ALE compatible we should do the points location before outputting the information. This can be easily achieved by moving the points location to a private method and introducing a `"update_output_points_at_each_time_step" : false` argument in the input settings.

Reply:
done :)